framework/src/meshgenerators/CutMeshByPlaneGenerator.C

```diff
@@ -0,0 +1,458 @@
+
+  ReplicatedMesh & mesh = *replicated_mesh_ptr;
+
+  // Subdomain ID for new utility blocks must be new
+  std::set<subdomain_id_type> subdomain_ids_set;
+  mesh.subdomain_ids(subdomain_ids_set);
+  const subdomain_id_type max_subdomain_id = *subdomain_ids_set.rbegin();
+  const subdomain_id_type block_id_to_remove = max_subdomain_id + 1;
+
+  // For the boolean value in the pair, true means the element is crossed by the cutting plane
+  // false means the element is on the remaining side
+  std::vector<std::pair<dof_id_type, bool>> cross_and_remained_elems_pre_convert;
+
+  for (auto elem_it = mesh.active_elements_begin(); elem_it != mesh.active_elements_end();
+       elem_it++)
+  {
+    const unsigned int & n_vertices = (*elem_it)->n_vertices();
+    unsigned short elem_vertices_counter_1(0);
+    unsigned short elem_vertices_counter_2(0);
+    for (unsigned int i = 0; i < n_vertices; i++)
+    {
+      if (pointPlaneRelation((*(*elem_it)->node_ptr(i)), _plane_point, _plane_normal) ==
+          PointPlaneRelationIndex::plane_normal_side)
+        elem_vertices_counter_1++;
+      if (pointPlaneRelation((*(*elem_it)->node_ptr(i)), _plane_point, _plane_normal) !=
+          PointPlaneRelationIndex::opposite_plane_normal_side)
+        elem_vertices_counter_2++;
+    }
+    if (elem_vertices_counter_2 == n_vertices)
+      (*elem_it)->subdomain_id() = block_id_to_remove;
+    else
+    {
+      // Check if any elements to be processed are not first order
+      if ((*elem_it)->default_order() != Order::FIRST)
+        mooseError("Only first order elements are supported for cutting.");
+      cross_and_remained_elems_pre_convert.push_back(
+          std::make_pair((*elem_it)->id(), elem_vertices_counter_1 > 0));
+    }
+  }
+
+  std::vector<dof_id_type> converted_elems_ids_to_cut;
+  std::vector<dof_id_type> converted_elems_ids_to_retain;
+  // Then convert these non-TET4 elements into TET4 elements
+  MooseMeshUtils::convert3DMeshToAllTet4(mesh,
+                                         cross_and_remained_elems_pre_convert,
+                                         converted_elems_ids_to_cut,
+                                         converted_elems_ids_to_retain,
+                                         block_id_to_remove,
+                                         false);
+
+  std::vector<Node *> new_on_plane_nodes;
+  // Cut the TET4 Elements
+  for (const auto & converted_elems_id_to_cut : converted_elems_ids_to_cut)
+  {
+    tet4ElemCutter(mesh,
+                   converted_elems_id_to_cut,
+                   _plane_point,
+                   _plane_normal,
+                   block_id_to_remove,
+                   new_on_plane_nodes);
+  }
+
+  // Delete the block to remove
+  for (auto elem_it = mesh.active_subdomain_elements_begin(block_id_to_remove);
+       elem_it != mesh.active_subdomain_elements_end(block_id_to_remove);
+       elem_it++)
+    mesh.delete_elem(*elem_it);
+
+  mesh.contract();
+  mesh.prepare_for_use();
```

Comment:
```suggestion
  mesh.set_isnt_prepared();
```
let s try this. We seek to prepare as rarely as possilbe